File: RedSheet.DbEntities/Models/PowerType.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("PowerTypes", Schema = "dbo")]
	[Serializable]
	public partial class PowerType  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int PowerTypeId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string PowerTypeName { get; set; } 
	[InverseProperty("PowerType")]
	public ICollection<PowerTypeDetail> PowerTypeDetails { get; set; } 


	public PowerType()
	{
	this.PowerTypeDetails = new HashSet<PowerTypeDetail>();

    }

    }
}

File: redsheet-portal/src/app/database-models/power-type.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { PowerTypeDetail,  } from './'
export class PowerType {
    constructor(powerType?: PowerType )  {
        let properties = [ "powerTypeId", "powerTypeName", "powerTypeDetails",];
        for (let property of properties)
            if (powerType && powerType[property])
                this[property] = powerType[property];
    }
 
	powerTypeId : number =   0 ;
 
    @required()
    @maxLength(200)
	powerTypeName : string =   undefined;
	@nested(PowerTypeDetail)
	powerTypeDetails: PowerTypeDetail[];



}
